### PR TITLE
full precision div

### PR DESCRIPTION
**Pull request by thedavidmeister**
## Checks

By submitting this for review, I'm confirming I've done the following:
- [ ] made this PR as small as possible
- [ ] unit-tested any new functionality
- [ ] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

## Summary by CodeRabbit

* **New Features**
  * Division/inverse use a consistent higher-precision scaling (effective scale -76) and a new public MulDivOverflow error is exposed.

* **Bug Fixes**
  * Reduced intermediate arithmetic overflow risk and improved handling of extreme signed-edge cases.

* **Tests**
  * Many test expectations, constants and some test names updated; new reciprocal/overflow tests added and gas snapshot metrics refreshed.

* **Documentation**
  * Clarified notes around inverse and pow10 behavior under the new precision model.

### CODE REVIEW

File: crates/float/src/lib.rs [hunk 1]
```diff
@@ -1368,7 +1368,11 @@ mod tests {
         let zero = Float::parse("0".to_string()).unwrap();
         let err = (one / zero).unwrap_err();
 
-        assert!(matches!(err, FloatError::Revert(_)));
+
```
Comment (thedavidmeister) [inline crates/float/src/lib.rs:1371]:
```suggestion
```

File: crates/float/src/lib.rs [hunk 2]
```diff
@@ -1368,7 +1368,12 @@ mod tests {
         let zero = Float::parse("0".to_string()).unwrap();
         let err = (one / zero).unwrap_err();
 
-        assert!(matches!(err, FloatError::Revert(_)));
+        dbg!(&err);
```
[the remark this answers is not in the document]
Comment (thedavidmeister) [reply crates/float/src/lib.rs]: fixed

**Review by hardyjosh — APPROVED** [review]
